libsplinter/src/rest_api/auth/allow_keys.rs

```diff
@@ -0,0 +1,401 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A file-backed authorization handler for defining admin keys
+
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+use std::time::SystemTime;
+
+use crate::error::InternalError;
+
+use cylinder::PublicKey;
+
+use super::{identity::Identity, AuthorizationHandler, AuthorizationHandlerResult};
+
+/// A file-backed authorization handler that permits admin keys
+///
+/// The authorization handler only accepts [`Identity::Key`] identities; if a different type of
+/// identity is provided, the handler will return [`AuthorizationHandlerResult::Continue`]. If a key
+/// is provided, the handler will check if the key is present in the backing file. If the key is in
+/// the backing file, the handler will return [`AuthorizationHandlerResult::Allow`]; if not, it will
+/// return [`AuthorizationHandlerResult::Continue`]. The `permission_id` argument for
+/// [`AuthorizationHandler::has_permission`] is ignored because this authorization handler provides
+/// admin privileges (all permissions).
+///
+/// The authorization handler's backing file must be a list of valid Cylinder [`PublicKey`]s,
+/// separated by newlines.
+///
+/// The list of keys in the file are cached in-memory by the authorization handler; this means that
+/// the handler will continue to work even if the backing file becomes unavailable. Each time the
```

Comment:
Interested in opinions on whether or not the handler should tolerate the file being removed

Reply:
In this auth case, it probably would make sense to treat it the same as an empty file.

Reply:
Yeah that makes more sense; updated.